src/provider/controller/ChannelHandler.ts

```diff
@@ -0,0 +1,131 @@
+import {injectable, inject} from 'inversify';
+
+import {Model} from '../model/Model';
+import {Inject} from '../common/Injectables';
+import {DEFAULT_CHANNEL_ID, ChannelId, FDC3Error, ChannelError, ChannelChangedEvent, Context} from '../../client/main';
+import {DesktopContextChannel, DefaultContextChannel, ContextChannel} from '../model/ContextChannel';
+import {AppWindow} from '../model/AppWindow';
+import {Signal1} from '../common/Signal';
+import {EventTransport} from '../../client/internal';
+
+const DESKTOP_CHANNELS = [
+    {
+        id: 'red',
+        name: 'Red',
+        color: 0xFF0000
+    },
+    {
+        id: 'orange',
+        name: 'Orange',
+        color: 0xFF8000
+    },
+    {
+        id: 'yellow',
+        name: 'Yellow',
+        color: 0xFFFF00
+    },
+    {
+        id: 'green',
+        name: 'Green',
+        color: 0x00FF00
+    },
+    {
+        id: 'blue',
+        name: 'Blue',
+        color: 0x0000FF
+    },
+    {
+        id: 'purple',
+        name: 'Purple',
+        color: 0xFF00FF
+    }
+];
+
+
+@injectable()
+export class ChannelHandler {
+    public readonly onChannelChanged: Signal1<EventTransport<ChannelChangedEvent>> = new Signal1<EventTransport<ChannelChangedEvent>>();
+
+    private readonly _model: Model;
+
+    constructor(@inject(Inject.MODEL) model: Model,) {
+        this._model = model;
+
+        this._model.onWindowAdded.add(this.onModelWindowAdded, this);
+        this._model.onWindowRemoved.add(this.onModelWindowRemoved, this);
+    }
+
+    public registerChannels(): void {
+        const defaultChannel = new DefaultContextChannel(DEFAULT_CHANNEL_ID);
+
+        this._model.registerChannel(defaultChannel);
+        for (const channel of DESKTOP_CHANNELS) {
+            this._model.registerChannel(new DesktopContextChannel(channel.id, channel.name, channel.color));
```

Comment:
Until we this is configurable, there isn't really a right place for this to live. I don't think Model owning the list is less wrong that ChannelHandler owning the list, and pulling it into it's own object - I'm not sure what that gains us. Could potentially just pull out a list the ChannelHandler reads and then puts into the Model. But like, this will all change when we implement this being configurable and this doesn't cause any problems until then.

Reply:
Changed. the trade-off is that Model is now aware of the specific logic of going from the config object to a `DefaultContextChannel`/`DesktopContextChannel`